src/reflect/scala/reflect/internal/util/SourceFile.scala

```diff
@@ -171,12 +171,30 @@ class BatchSourceFile(val file : AbstractFile, content0: Array[Char]) extends So
   }
 
   private lazy val lineIndices: Array[Int] = {
+    def countEOL(cs: Array[Char]): Int = {
+      var i, c = 0
+      while (i < cs.length){
+        if (isAtEndOfLine(i))
+          c += 1
+        i += 1
+      }
+      c
+    }
     def calculateLineIndices(cs: Array[Char]) = {
-      val buf = new ArrayBuffer[Int]
-      buf += 0
-      for (i <- 0 until cs.length) if (isAtEndOfLine(i)) buf += i + 1
-      buf += cs.length // sentinel, so that findLine below works smoother
-      buf.toArray
+      // count EOL characters in cs
+      val res = new Array[Int](countEOL(cs) + 2)
+      res(0) = 0
+      res(res.length - 1) = cs.length
+      // fill res with positions of EOL 
+      var i,j = 0
+      while(i < cs.length && j < res.length - 1){
```

Comment:
Let's keep the "// sentinel, so that findLine below works smoother" comment, and while making that change, could you please add a space between `){`  and `,j` :)

Reply:
I think that style actually follows from no-space-after-while, that is, space is optional before bracket. Maybe not `for{`; it's any bracket that looks like a function call. (I wore out my welcome already, so I'm only semi-serious, or `;`-serious.)

Reply:
@retronym I have applied your suggestions.

Reply:
Retronym suggests applications, and we apply his suggestions.